react-components/src/hooks/use3dScenes.tsx

```diff
@@ -33,56 +40,11 @@ export const use3dScenes = (
     try {
       const scenesQueryResult = await fdmSdk.queryNodesAndEdges(scenesQuery);
 
-      const scenesMap: Record<
-        Space,
-        Record<ExternalId, SceneData>
-      > = scenesQueryResult.items.sceneModels.reduce(
-        (acc, item) => {
-          const edge = item as EdgeItem;
-
-          const { space, externalId } = edge.startNode;
-
-          const properties = Object.values(
-            Object.values(edge.properties)[0] as Record<string, unknown>
-          )[0] as Cdf3dRevisionProperties;
-
-          const newModelId = Number(edge.endNode.externalId);
-          const newModelRevisionId = Number(properties?.revisionId);
-
-          if (isNaN(newModelId) || isNaN(newModelRevisionId)) {
-            return acc;
-          }
-
-          const newModel = createModelFromEdge(newModelId, newModelRevisionId, properties);
-
-          if (acc[space] === undefined) {
-            acc[space] = {};
-          }
-          const spaceMap = acc[space];
-
-          if (spaceMap[externalId] === undefined) {
-            spaceMap[externalId] = { name: '', modelOptions: [] };
-          }
-          const externalIdMap = spaceMap[externalId];
-          externalIdMap.modelOptions.push(newModel);
-
-          return acc;
-        },
-        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-        {} as Record<Space, Record<ExternalId, SceneData>>
-      );
-
-      scenesQueryResult.items.scenes.forEach((scene) => {
-        const { space, externalId } = scene;
-
-        if (scenesMap[space]?.[externalId] !== undefined) {
-          const properties = Object.values(
-            Object.values(scene.properties)[0] as Record<string, unknown>
-          )[0] as SceneConfigurationProperties;
-          scenesMap[space][externalId].name = properties.name;
-        }
-      });
+      const scenesMap = createMapOfScenes(scenesQueryResult.items.scenes);
+      populateSceneMapWithModels(scenesQueryResult.items.sceneModels, scenesMap);
+      populateSceneMapWith360Images(scenesQueryResult.items.scene360Collections, scenesMap);
 
+      console.log('scenesMap', scenesMap);
```

Comment:
:eyes:

Reply:
Good catch 😅 , fixed in [808e42b](https://github.com/cognitedata/reveal/pull/4131/commits/808e42b35074ad506aac0d3f47f0e32ad69050fa)